src/dlsproto/client/internal/SuspendableRequest.d

```diff
@@ -471,15 +471,11 @@ struct SuspendableRequest
         // state change in the delegate.
         auto last_state = this.shared_working.desired_state;
         ReceivedMessageAction msg_action;
-        int resume_code = this.conn.receiveAndHandleEvents(
-            ( in void[] received )
-            {
-                msg_action = handle_received_message(received);
-            }
-        );
+        auto event = this.conn.nextEvent(this.conn.NextEventFlags.Receive);
 
-        if ( resume_code < 0 ) // Fiber resumed due to received data
+        if ( event.active == event.active.received )
```

Comment:
No `if` required, I think. The only event type that can ever be returned is `received`.

Reply:
Oh, right, I see - it enforces that the right thing has been done internally (no `Send` is possible if we have not intended sending).

Reply:
But bear in mind that `nextEvent` can always return a `resumed` event.